django/contrib/admin/options.py

```diff
@@ -143,7 +143,7 @@ def formfield_for_dbfield(self, db_field, request, **kwargs):
                 kwargs = dict(self.formfield_overrides[db_field.__class__], **kwargs)
 
             # Get the correct formfield.
-            if isinstance(db_field, models.ForeignKey):
+            if db_field.many_to_one:
                 formfield = self.formfield_for_foreignkey(db_field, request, **kwargs)
```

Comment:
I agree. I think it will be safer to leave this as `isinstance(db_field, models.ForeignKey)`. I'd imagine custom many_to_one=True fields will have their own widget, and will not want this specific behavior.